### PR TITLE
First steps to get an m1 runner role built

### PR DESCRIPTION
**Pull request by pmaslana**

### CODE REVIEW

File: m1-runner/tasks/main.yml [hunk 1]
```diff
@@ -1,15 +1,56 @@
...
-    curl -O -L https://github.com/actions/runner/releases/download/v"{{ runner_release_version }}"/actions-runner-osx-x64-"{{ runner_release_version }}".tar.gz
-    tar xzf ./actions-runner-osx-x64-"{{ runner_release_version }}".tar.gz
-    ./config.sh --url https://github.com/Chia-Network --token ATAQBLQ36XEKM6ED6XQ7FNDBHE6ZM --runnergroup "M1 Runners" --labels macOS,self-hosted,X64,m1 --name m1-01 --work _work
-  become_user: chia-network
+#This checks to see if the service was installed. It is installed by running
+# ./svc.sh install in the actions-runner directory.
+- name: Check for the auto-generated service file
+  stat:
+    path: /Library/LaunchAgents/actions.runner.Chia-Network.m1-01.plist
+  register: runner_file
```
Comment (cmmarslender) [inline m1-runner/tasks/main.yml]:
```suggestion
```

File: m1-runner/tasks/main.yml [hunk 2]
```diff
@@ -1,15 +1,56 @@
...
-  become: chia-network
-  shell: "{{ path_to_service }}/svc.sh install"
+- name: Insert the proper path for gnu-tar in .zshenv and source the file.
+  lineinfile:
+    path: "~/.zshenv"
+    line: export PATH="/opt/homebrew/opt/gnu-tar/libexec/gnubin:$PATH"
+    insertbefore: EOF
+
+- name: Make action-runner directory
+  when: not runner_file.stat.exists
```
Comment (cmmarslender) [inline m1-runner/tasks/main.yml]:
```suggestion
```

**Review by cmmarslender — CHANGES_REQUESTED** [review]

File: m1-runner/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+---
+# chia-runner/tasks/main.yml
+
+- name: Insert the proper path for gnu-tar in .zshenv.
+  lineinfile:
+    path: "~/.zshenv"
+    line: export PATH="/opt/homebrew/opt/gnu-tar/libexec/gnubin:$PATH"
+    insertbefore: EOF
+
+- name: Source the ~/.zshenv file
```
Comment (cmmarslender) [inline m1-runner/tasks/main.yml:10]: Have you tried without this? Once we identified the correect file, this should just be sourced automatically

File: m1-runner/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    mode: '0755'
+
+- name: Download runner tarball
+  get_url:
+    url: https://github.com/actions/runner/releases/download/v{{runner_release_version}}/actions-runner-osx-x64-{{runner_release_version}}.tar.gz
+    dest: "{{ path_to_service }}"
+    mode: '0755'
+
+- name: Unarchive runner tarball
+  when: not service_script.stat.exists
```
Comment (cmmarslender) [inline m1-runner/tasks/main.yml:26]:
```suggestion
```

File: m1-runner/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  ansible.builtin.unarchive:
+    src: "{{ path_to_service }}/actions-runner-osx-x64-{{runner_release_version}}.tar.gz"
+    dest: "{{ path_to_service }}"
+    remote_src: yes
+    list_files: yes
+    creates: "{{ path_to_service }}/actions-runner/svc.sh"
+
+- name: Configure the runner for GitHub
+  when: not service_script.stat.exists
```
Comment (cmmarslender) [inline m1-runner/tasks/main.yml:35]:
```suggestion
```

File: m1-runner/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  shell: ./config.sh --url https://github.com/Chia-Network --token {{registration_token}} --runnergroup "M1 Runners" --labels m1 --name m1-01 --unattended
+  args:
+    chdir: "{{ path_to_service }}"
+    creates: "{{ path_to_service }}/actions-runner/svc.sh"
+
+- name: Install the runner service
```
Comment (cmmarslender) [inline m1-runner/tasks/main.yml:41]: is there a creates or else a condition we can use to only run the "install" when we need to? What happens if we run install and its already installed?

**Review by cmmarslender — CHANGES_REQUESTED** [review]

File: m1-runner/tasks/main.yml (hunk 3 above)
Comment (pmaslana) [reply m1-runner/tasks/main.yml:10]: Sourced automatically after a line has been appended to it?

Comment (cmmarslender) [reply m1-runner/tasks/main.yml:10]: Oh hmm. Thats a good question - not sure if its loading that for every command that gets run or not - may not actually be, though if you end up testing, would be curious to know the result, but this is fine as is then

**Review by cmmarslender — DISMISSED** [review]

**Review by cmmarslender — APPROVED** [review]

**Pull request merged**